lib/puppet/provider/package/windows.rb

```diff
@@ -64,7 +64,7 @@ def install
 
     command = [installer.install_command(resource), install_options].flatten.compact.join(' ')
     working_dir = File.dirname(resource[:source])
-    if !Puppet::FileSystem.exist?(working_dir) && resource[:source] =~ /\.msi"?\Z/i
+    if !Puppet::FileSystem.exist?(working_dir)
```

Comment:
can also use unless instead of negating the condition

Reply:
Reminder for tomorrow: Check if it's ok for `&& resource[:source] =~ /\.msi"?\Z/i` to be missing here.

Reply:
If I remembered correctly, I removed it because it was setting `working_dir = nil` when installing a .msi from URL, and now, we can also install .exe from URL, so and adding `resource[:source] =~ /\.msi"?\Z/i || resource[:source] =~ /\.exe"?\Z/i` would always be true as either a .exe or a .msi is acceptable